### PR TITLE
Fix #18199 : Add fallbacks for confirmation title on NFT transfers

### LINKED ISSUE #18199: [Bug]: NFT details are not displayed
### Describe the bug

The following NFT on Polygon does not display the collection name, on the confirmation page.
- The title icon displays an `identicon` and not the collection image
- The title displays `0` and not and not the collection name
- The gas detail displays `null` and not the collection name

Screenshot 2023-03-16 at 16 19 06

The following NFT on Polygon does not display the collection image, on the home page or the asset page.

unknown collection

AppStateController
```
		"nftsDropdownState": {
			"0x56a355d3427bc2b1e22c78197af091230919cc2a": {
				"0x89": {
					"0x8577010864d3218F1f86d307cce6467eB2dBc066": true
				}
			}
		},
```

NftController
```
	"NftController": {
		"allNftContracts": {
			"0x56a355d3427bc2b1e22c78197af091230919cc2a": {
				"137": [{
					"address": "0x8577010864d3218F1f86d307cce6467eB2dBc066",
					"name": "gho.lens-Collect-1",
					"symbol": "gho.-Cl-1"
				}]
			}
		},
		"allNfts": {
			"0x56a355d3427bc2b1e22c78197af091230919cc2a": {
				"137": [{
					"address": "0x8577010864d3218F1f86d307cce6467eB2dBc066",
					"tokenId": "132027",
					"favorite": false,
					"isCurrentlyOwned": true,
					"name": null,
					"description": null,
					"image": null,
					"standard": "ERC721"
				}]
			}
		},
		"ignoredNfts": []
	}
```

Links to the following NFT https://opensea.io/assets/matic/0x8577010864d3218F1f86d307cce6467eB2dBc066/132027

### Steps to reproduce

1. Switch to Polygon
2. Import the above NFT
3. Block the request for the metadata see the video below

https://user-images.githubusercontent.com/53189696/226927809-cc4c11b4-8b5a-4ef2-a4d3-d72c5288f27d.mov

### Error messages or log output

_No response_

### Version

10.28

### Build type

None

### Browser

Chrome, Firefox

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by adonesky1**
## Explanation
Fixes: #18199 

The bug was filed originally because antivirus software was blocking the URL of arweave.  This PR provides a fallback name and total subtitle based on the collection name.

### With arweave address blocked
ConfirmDisplay

### With requests answered normally
NormalNFTSend

### CODE REVIEW

File: ui/pages/confirm-transaction-base/confirm-transaction-base.container.js [hunk 1]
```diff
@@ -185,9 +186,11 @@ const mapStateToProps = (state, ownProps) => {
   );
 
   const isNftTransfer = Boolean(
-    allNftContracts?.[selectedAddress]?.[chainId]?.find((contract) => {
-      return isEqualCaseInsensitive(contract.address, fullTxData.txParams.to);
-    }),
+    allNftContracts?.[selectedAddress]?.[hexToDecimal(chainId)]?.find(
```
Comment (danjm) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.container.js:189]: Why is this change necessary?

Comment (danjm) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.container.js:189]: has `chainId` changed to a hex value? or was this buggy before?

Comment (adonesky1) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.container.js:189]: This was buggy before... its an unfortunate effect of the situation where the two network controllers (extension and core - which will be merged soon!) represent the chainId differently (extension as a hex, and core as a decimal). This results in a bunch of other controllers from core expecting chainIds as decimals. This includes `NftController`, so any time we read state from this controller we have to convert the chainId to decimal. related: https://github.com/MetaMask/metamask-extension/pull/17384

**Review by danjm — APPROVED** [review]
LGTM

**Review by pedronfigueiredo — APPROVED** [review]

**Comment by PeterYinusa** [comment]
Can we please add some automated tests for the fallback logic?

**Review by danjm — CHANGES_REQUESTED** [review]
Removing my approval to second Peter's request for tests

**Review by danjm — APPROVED** [review]

**Comment by danjm** [comment]
David is working on the tests separately

**Pull request merged**